### DISCUSSION TITLE
Register and do not use email activate and test forgot password

### DISCUSSION THREAD

**Discussion by jlopes90** [codeigniter4/shield#279] [Q&A]
"Register with email activate"

Register and do not use email activate and test forgot password then message appears

> CodeIgniter\Shield\Exceptions\LogicException
> 
> The user has identities for action, so cannot complete login. If you want to start to login with auth action, use startLogin() instead. Or delete identities for action in database. user_id: 1

I know it's an exaggeration, which can happen email activate accidentally deleted and don't remember filling in the password or something.

**Comment by kenjis** [comment]
What's your question?

If the account is not activated, the user must activate it first.

**Reply by MGatner** [reply]
I _think_ he means this:
1. User registers
2. Email activation is sent
3. Before activating, user clicks "Forgot password"
4. Controller throws exception

I'll look at the code to confirm, but if that's the case we should be catching that exception and offering a helpful message to the user.

**Reply by MGatner** [reply]
Yes this is precisely it. When `MagicController::verify()` tries to log the user in:
```php
// Log the user in
$authenticator->loginById($identity->user_id);
```
Probably the fix is actually not to send a magic link if the registration isn't fully complete.
